service/src/main/frontend/src/functions/formatters.ts

```diff
@@ -0,0 +1,106 @@
+/* eslint-disable no-var */
+import { Ref, WritableComputedRef, computed, defineComponent, ref } from 'vue';
+
+// Parses a Value to remove uncessessary 'undefined' or empty values.
+export function parseVal(colName: any, colValue: any): string {
+  if (colName === 'Types' || colName === 'Descriptions') {
+    if (colValue == undefined) {
+      return '';
+    } else {
+      return colValue.toString();
+    }
+  } else {
+    return colValue;
+  }
+}
+
+// Produces the max substring for the table, ads '...' if above 34 chars.
+export function maxSubstring(str: any, colName: any): any {
+  if (str == undefined) {
+    return;
+  } else if ((colName === 'fieldName' || colName === 'internalFieldName') && str.length > 32) {
+    return str.substring(0, 30) + ' ...';
+  } else if ((colName === 'Types') && str.length > 14) {
+    return str.substring(0, 9) + ' ...';
+  } else if ((colName === 'Descriptions') && str.length > 24) {
+    return str.substring(0, 22) + ' ...';
+  } else if ((colName === 'CopyPaste') && str.length > 42) {
+    return str.substring(0, 40) + ' ...';
+  } else {
+    return str;
+  }
+}
+
+// Defines how the expandability is parsed on the table.
+export function buttonParse(col: any, row: any): boolean {
+  return row.button == 1;
+}
+
+// Toggles how the row collapses based on the DOM. Filters visible rows.
+export function toggleVisibility(row: any) {
+  row.toggleVisibility();
+}
+
+// Set the Visibility in DOM, sorts and filters by lastUpdated, and the respective row to render button.
+export function setVisibility(rows: readonly any[]) {
+  const fieldVisibility: Map<string, Ref<boolean>> = new Map<
+    string,
+    Ref<boolean>
+  >();
+  const buttonValues: Map<string, number> = new Map<string, number>();
+
+  for (var row of rows) {
```

Comment:
Just as a signal to the code reviewer, let's use `const` here to indicate you're not intending to change the reference to the variable.
```suggestion
  for (const row of rows) {
```